# Práctica 1.1 — Probabilidad y Estadística

## Análisis de la evolución poblacional mundial

## Objetivo

Utilizar un conjunto de datos real para comprobar el funcionamiento del entorno de análisis de datos y aplicar medidas estadísticas descriptivas básicas utilizando Pandas y NumPy.

## Actividad 1 — Identificación de la fuente

**Propósito:** Reconocer la procedencia del conjunto de datos antes de utilizarlo y documentar correctamente su fuente.

Los datos utilizados en esta práctica provienen de **Our World in Data (OWID)**.

La fuente original es **United Nations, World Population Prospects (2024)**.

**Cita:**

> UN, World Population Prospects (2024) – processed by Our World in Data. "Annual change in population – UN WPP" [dataset]. United Nations, "World Population Prospects"; United Nations, "World Population Prospects - Interim Update" [original data].

**Preguntas**

1. **¿Quién produjo originalmente los datos?**
   Los datos originales los produce la ONU, específicamente a través de su reporte *World Population Prospects (2024)*, que es la referencia mundial en proyecciones y estimaciones demográficas.

2. **¿Qué organización los procesa para su utilización en Our World in Data?**
   Es el propio equipo de Our World in Data (OWID) quien toma esos datos de la ONU y los limpia, estandariza y adapta para que puedan usarse en sus gráficos y bases de datos públicas.

3. **¿Qué representa la variable que analizarás?**
   La variable `Annual change in population` muestra cuánto cambió la población de un país o región de un año a otro (en número de personas). Es un cambio *neto*, es decir, ya incluye el efecto combinado de nacimientos, muertes y migración, no solo uno de esos factores por separado.

## Actividad 2 — Carga y exploración de datos

**Propósito:** Comprobar que el entorno de Python, Pandas y Jupyter funciona correctamente y realizar una primera exploración del dataset.

### 2.1 Importar las bibliotecas

In [1]:
import pandas as pd
import numpy as np

### 2.2 Cargar el archivo CSV

In [2]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

### 2.3 Explorar el DataFrame

In [3]:
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
df.shape

(38400, 5)

In [5]:
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [6]:
df.dtypes

Entity                                   object
Code                                     object
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [7]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [8]:
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [9]:
df["Entity"].nunique()

256

**Preguntas**

1. **¿Cuántas filas y columnas tiene el dataset?**
   El DataFrame tiene 38,400 filas y 5 columnas, según se ve en `df.shape`.

2. **¿Qué tipo de datos contiene la variable Year?**
   Es de tipo `int64`, es decir, un número entero. Tiene sentido, ya que cada año se representa como un valor entero (por ejemplo, 1995, 2010, 2023).

3. **¿Existen valores faltantes?**
   Sí, y bastantes. La columna `Code` tiene 1,200 valores nulos (esto pasa porque algunas entidades son agregados regionales, como "Africa (UN)" o "World", que no tienen un código ISO de país). Además, tanto `Annual change in population` (19,712 nulos) como `Annual population change (Projected)` (18,688 nulos) tienen huecos importantes, lo cual es esperable: el dataset combina en las mismas columnas el periodo histórico y el proyectado, así que para cada fila solo una de las dos columnas suele tener dato y la otra queda vacía.

4. **¿Cuántas entidades diferentes contiene el dataset?**
   Hay 256 entidades distintas, entre países, regiones y agrupaciones (como continentes o clasificaciones por nivel de desarrollo).

## Actividad 3 — Estadística descriptiva

**Propósito:** Aplicar medidas estadísticas básicas a una variable poblacional.

**Instrucciones**

Se selecciona un año del periodo histórico (2023, el más reciente disponible) y se filtran los datos correspondientes. Sobre la variable de **cambio anual de población** (`Annual change in population`) se calculan las medidas estadísticas descriptivas.

In [10]:
anio = 2023
df_anio = df[df["Year"] == anio]
df_anio.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
72,Afghanistan,AFG,2023,875915.0,NaN
222,Africa (UN),UN_AFR,2023,33886868.0,NaN
372,Albania,ALB,2023,-15954.0,NaN
522,Algeria,DZA,2023,686831.0,NaN
672,American Samoa,ASM,2023,-821.0,NaN


In [11]:
variable = "Annual change in population"

media = df_anio[variable].mean()
mediana = df_anio[variable].median()
varianza = df_anio[variable].var()
desviacion_estandar = df_anio[variable].std()
minimo = df_anio[variable].min()
maximo = df_anio[variable].max()

print("Media:", media)
print("Mediana:", mediana)
print("Varianza:", varianza)
print("Desviación estándar:", desviacion_estandar)
print("Mínimo:", minimo)
print("Máximo:", maximo)

Media: 2008097.765625
Mediana: 46865.5
Varianza: 82427560967665.2
Desviación estándar: 9078962.549083745
Mínimo: -3315935.0
Máximo: 72315107.0


In [12]:
entidad_min = df_anio.loc[df_anio[variable].idxmin(), "Entity"]
entidad_max = df_anio.loc[df_anio[variable].idxmax(), "Entity"]

print("Entidad con el valor mínimo:", entidad_min)
print("Entidad con el valor máximo:", entidad_max)

Entidad con el valor mínimo: Ukraine
Entidad con el valor máximo: Less developed regions, excluding China


**Preguntas**

1. **¿Cuál es el valor de la media?**
   Para 2023, la media del cambio anual de población es de aproximadamente 2,008,098 personas. Este número está bastante inflado porque el dataset mezcla países individuales con agregados regionales (como continentes), que suman muchísima más gente que un país promedio.

2. **¿Cuál es el valor de la mediana?**
   La mediana es de 46,865.5 personas, mucho más baja que la media. Esa diferencia tan grande entre media y mediana es una señal clara de que la distribución está muy sesgada por esos valores extremos de las regiones agregadas.

3. **¿Cuál es el valor de la desviación estándar?**
   La desviación estándar es de aproximadamente 9,078,963 personas, un valor enorme que confirma la gran dispersión de los datos: hay desde países que pierden población hasta regiones enteras que ganan decenas de millones de habitantes en un solo año.

4. **¿Qué entidad presenta el valor mínimo y cuál el máximo?**
   Ucrania (Ukraine) presenta el valor mínimo, con una caída de población de más de 3.3 millones de personas en 2023, probablemente reflejo del conflicto y la migración forzada de esos años. El valor máximo corresponde a "Less developed regions, excluding China", una categoría agregada de la ONU que agrupa a muchos países en desarrollo, por lo que su crecimiento (más de 72 millones) es la suma de decenas de naciones y no el de un solo país.

## Actividad 4 — Comprobación del entorno

**Propósito:** Comprobar que el notebook está utilizando correctamente las herramientas configuradas para la práctica.

In [13]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.13.5 | packaged by Anaconda, Inc. | (main, Jun 12 2025, 11:23:37) [Clang 14.0.6 ]
Pandas: 2.2.3
NumPy: 2.1.3


Se verifica que las versiones anteriores corresponden al entorno virtual (`.venv`) configurado para esta práctica.